### ISSUE TITLE
`disableStopDetection: true` does not disable stop detection on Android

### ISSUE THREAD

**Issue by nattyg93** [transistorsoft/flutter_background_geolocation#67]
## Your Environment
* Plugin version: 1.0.7
* Platform: Android
* OS version: Pie
* Device manufacturer / model: Multiple devices, including (among others) Xiami Mi A2, Samsung Galaxy S9, Nokia 6.1
* Flutter info (`flutter doctor`): 
```
[✓] Flutter (Channel stable, v1.5.4-hotfix.2, on Mac OS X 10.14.5 18F132, locale en-AU)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
[✓] Android Studio (version 3.2)
[✓] VS Code (version 1.34.0)
[✓] Connected device (1 available)

• No issues found!
```
* Plugin config:
```dart
      bg.Config(
        desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
        distanceFilter: 1.0,
        stopOnTerminate: false,
        startOnBoot: true,
        foregroundService: true,
        isMoving: true,
        disableStopDetection: true,
        pausesLocationUpdatesAutomatically: false,
        debug: false,
        logLevel: bg.Config.LOG_LEVEL_VERBOSE,
        url: "URL_GOES_HERE",
        autoSync: true,
        autoSyncThreshold: 50,
        batchSync: true,
        maxBatchSize: 50,
      ),
```

## Expected Behavior
The docs state for disableStopDetection:
``` dart
/// Location-services **will never turn OFF** if you set this to **`true`**!  It will be purely up to you or the user to execute [BackgroundGeolocation.changePace]:false or [BackgroundGeolocation.stop] to turn off location-services.
```

I would expect that the plugin does not suspend after 5 minutes and locations services remain on when disableStopDetection is set to true.

## Actual Behavior
As can be clearly seen in the logs, the plugin stops when stationary for the default 5 minutes and location services turn off.

## Steps to Reproduce

1. Start the plugin and supply the config.
2. Leave the phone stationary for 5 minutes.
3. The plugin will suspend and location services turn off.

## Context
We are building a research app that should accurately tracks participants' movements, where the delay caused by the activity change to start location services is unacceptable.

## Debug logs

<details>
	<summary>Logs</summary>

```

{
    "activityRecognitionInterval": 10000,
    "allowIdenticalLocations": false,
    "autoSync": true,
    "autoSyncThreshold": 50,
    "batchSync": true,
    "debug": false,
    "deferTime": 0,
    "desiredAccuracy": -1,
    "desiredOdometerAccuracy": 100,
    "disableElasticity": false,
    "disableStopDetection": true,
    "distanceFilter": 1,
    "elasticityMultiplier": 1,
    "enableHeadless": false,
    "enableTimestampMeta": false,
    "extras": {},
    "fastestLocationUpdateInterval": -1,
    "forceReloadOnBoot": false,
    "forceReloadOnGeofence": false,
    "forceReloadOnHeartbeat": false,
    "forceReloadOnLocationChange": false,
    "forceReloadOnMotionChange": false,
    "forceReloadOnSchedule": false,
    "foregroundService": true,
    "geofenceInitialTriggerEntry": true,
    "geofenceModeHighAccuracy": false,
    "geofenceProximityRadius": 1000,
    "geofenceTemplate": "",
    "headers": {},
    "headlessJobService": "com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask",
    "heartbeatInterval": -1,
    "httpRootProperty": "location",
    "httpTimeout": 60000,
    "isMoving": false,
    "locationTemplate": "",
    "locationTimeout": 60,
    "locationUpdateInterval": 1000,
    "locationsOrderDirection": "ASC",
    "logLevel": 5,
    "logMaxDays": 3,
    "maxBatchSize": 50,
    "maxDaysToPersist": 1,
    "maxRecordsToPersist": -1,
    "method": "POST",
    "minimumActivityRecognitionConfidence": 75,
    "notification": {
        "layout": "",
        "title": "",
        "text": "Location Service activated",
        "color": "",
        "channelName": "TSLocationManager",
        "smallIcon": "",
        "largeIcon": "",
        "priority": 0,
        "strings": {},
        "actions": []
    },
    "params": {},
    "persist": true,
    "persistMode": 2,
    "schedule": [],
    "scheduleUseAlarmManager": false,
    "speedJumpFilter": 300,
    "startOnBoot": true,
    "stationaryRadius": 25,
    "stopAfterElapsedMinutes": 0,
    "stopOnStationary": false,
    "stopOnTerminate": false,
    "stopTimeout": 5,
    "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
    "url": "URL_GOES_HERE",
    "useSignificantChangesOnly": false,
    "enabled": true,
    "schedulerEnabled": false,
    "trackingMode": 1,
    "odometer": 0,
    "isFirstBoot": false
}

05-29 10:03:40.362 DEBUG [TSConfig c] ℹ️   Persist config, dirty: [autoSyncThreshold, batchSync, desiredAccuracy, disableStopDetection, distanceFilter, headlessJobService, isMoving, logLevel, maxBatchSize, params, startOnBoot, stopOnTerminate, url]
05-29 10:03:40.370 INFO [HttpService flush] 
╔═════════════════════════════════════════════
║ HTTP Service (count: 43)
╠═════════════════════════════════════════════

05-29 10:03:40.383 INFO [BackgroundTaskManager onStartJob] ⏳ startBackgroundTask: 8
05-29 10:03:40.412 DEBUG [b allWithLocking] 
  ✅  Locked 43 records
05-29 10:03:40.415 INFO [HttpService a] 
  🔵  HTTP POST batch (43)
05-29 10:03:40.445 DEBUG [b a] 
  ℹ️  LocationAuthorization: Permission granted
05-29 10:03:40.452 INFO [HeartbeatService a] 
  🔴  Stop heartbeat
05-29 10:03:40.458 INFO [TSGeofenceManager start] 
  🎾  Start monitoring geofences
05-29 10:03:40.461 DEBUG [TSGeofenceManager stopMonitoringStationaryRegion] 
  🔴  Stop monitoring stationary region
05-29 10:03:40.469 DEBUG [b a] 
  ℹ️  LocationAuthorization: Permission granted
05-29 10:03:40.476 INFO [TrackingService a] 
  🔵  setPace: true → true
05-29 10:03:40.477 INFO [TrackingService f] 
╔═════════════════════════════════════════════
║ TrackingService motionchange: true
╠═════════════════════════════════════════════

05-29 10:03:40.477 INFO [ActivityRecognitionService a] 
  🎾  Start motion-activity updates
05-29 10:03:40.482 DEBUG [HttpService startMonitoringConnectivityChanges] 
  🎾  Start monitoring connectivity changes
05-29 10:03:40.489 DEBUG [a c] 
  🎾  Start monitoring powersave changes
05-29 10:03:40.492 DEBUG [HttpService a] 
╔═════════════════════════════════════════════
║ 📶  Connectivity change: connected? true
╠═════════════════════════════════════════════

05-29 10:03:40.495 INFO [HttpService flush] 
  ℹ️  HttpService is busy
05-29 10:03:40.536 DEBUG [ActivityRecognitionService a] 
  🚘 ️DetectedActivity [type=STILL, confidence=100]
05-29 10:03:40.549 DEBUG [ActivityRecognitionService onDestroy] 
  🔴  ActivityRecognitionService destroyed
05-29 10:03:40.619 INFO [LocationRequestService b] 
  ℹ️  Location availability: true
05-29 10:03:41.354 WARN [HttpService$b onResponse] 
  ⚠️  Response: 404, Not Found
05-29 10:03:41.385 DEBUG [b unlock] 
  ✅  UNLOCKED (43)
05-29 10:03:41.390 INFO [BackgroundTaskManager$Task stop] ⏳ stopBackgroundTask: 8
05-29 10:03:41.405 INFO [LocationRequestService a] 
╔═════════════════════════════════════════════
║ motionchange LocationResult: 8
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884529,147.329827 hAcc=17 et=+8d19h53m2s983ms alt=10.800000190734863 vAcc=4 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 58ms, time: 1559088221346

05-29 10:03:41.411 INFO [TSLocationManager onSingleLocationResult] 
  🔵  Acquired motionchange position, isMoving: true
05-29 10:03:41.414 DEBUG [TSLocationManager calculateMedianAccuracy] Median accuracy: 12.233
05-29 10:03:41.429 INFO [TSLocationManager requestLocationUpdates] 
  🎾  Location-services: ON
05-29 10:03:41.433 DEBUG [TSConfig translateDesiredAccuracy] translateDesiredAccuracy (true): -1
05-29 10:03:41.436 DEBUG [BackgroundTaskService onDestroy] 
05-29 10:03:41.438 INFO [b persist] 
  ✅  INSERT: 86159acb-af11-40bf-8dcd-ba9e967b6bc7
05-29 10:03:41.444 INFO [HttpService flush] 
╔═════════════════════════════════════════════
║ HTTP Service (count: 44)
╠═════════════════════════════════════════════

05-29 10:03:41.446 INFO [TSLocationManager removeLocationUpdates] 
  🔴  Location-services: OFF
05-29 10:03:41.448 INFO [TSLocationManager requestLocationUpdates] 
  🎾  Location-services: ON
05-29 10:03:41.451 DEBUG [TSConfig translateDesiredAccuracy] translateDesiredAccuracy (true): -1
05-29 10:03:41.460 INFO [BackgroundTaskManager onStartJob] ⏳ startBackgroundTask: 9
05-29 10:03:41.486 DEBUG [b allWithLocking] 
  ✅  Locked 44 records
05-29 10:03:41.489 INFO [HttpService a] 
  🔵  HTTP POST batch (44)
05-29 10:03:41.507 DEBUG [TrackingService a] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884529,147.329827 hAcc=17 et=+8d19h53m3s98ms alt=10.800000190734863 vAcc=4 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 44ms, time: 1559088221460

05-29 10:03:41.509 DEBUG [TSLocationManager onLocationResult] 
╔═════════════════════════════════════════════
║ Process LocationResult
╠═════════════════════════════════════════════

05-29 10:03:41.512 DEBUG [TSLocationManager onLocationResult] 
  ℹ️  IGNORED: same as last location
05-29 10:03:41.515 INFO [TSScheduleManager oneShot] 
  ⏰ Scheduled OneShot: STOP_TIMEOUT in 300000ms (jobID: 2059034116)
05-29 10:03:41.524 INFO [TrackingService a] 
  ℹ️  Distance from stoppedAtLocation: -33.608
05-29 10:03:41.534 INFO [TrackingService b] 
  ℹ️  Location availability: true
05-29 10:03:41.546 DEBUG [TrackingService a] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884529,147.329827 hAcc=17 et=+8d19h53m3s131ms alt=10.800000190734863 vAcc=4 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 51ms, time: 1559088221493

05-29 10:03:41.572 DEBUG [TSLocationManager onLocationResult] 
╔═════════════════════════════════════════════
║ Process LocationResult
╠═════════════════════════════════════════════

05-29 10:03:41.576 DEBUG [TSLocationManager onLocationResult] 
  ℹ️  IGNORED: same as last location
05-29 10:03:41.579 INFO [TrackingService a] 
  ℹ️  Distance from stoppedAtLocation: -33.608
05-29 10:03:42.204 WARN [HttpService$b onResponse] 
  ⚠️  Response: 404, Not Found
05-29 10:03:42.215 DEBUG [b unlock] 
  ✅  UNLOCKED (44)
05-29 10:03:42.219 INFO [BackgroundTaskManager$Task stop] ⏳ stopBackgroundTask: 9
05-29 10:03:42.223 DEBUG [BackgroundTaskService onDestroy] 
05-29 10:03:51.048 DEBUG [TrackingService a] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884527,147.329812 hAcc=10 et=+8d19h53m12s654ms alt=10.800000190734863 vel=0.0 vAcc=4 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 26ms, time: 1559088232000

05-29 10:03:51.054 DEBUG [TSLocationManager onLocationResult] 
╔═════════════════════════════════════════════
║ Process LocationResult
╠═════════════════════════════════════════════

05-29 10:03:51.059 DEBUG [TSLocationManager calculateMedianAccuracy] Median accuracy: 10.048
05-29 10:03:51.072 INFO [TrackingService a] 
  ℹ️  Distance from stoppedAtLocation: -25.624226
05-29 10:03:51.074 INFO [b persist] 
  ✅  INSERT: 4255453d-4f32-4531-b435-c16e9bec6301
05-29 10:03:53.302 DEBUG [TrackingService a] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884527,147.329796 hAcc=10 et=+8d19h53m14s871ms alt=10.800000190734863 vel=0.0 vAcc=4 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 66ms, time: 1559088233233

05-29 10:03:53.305 DEBUG [TSLocationManager onLocationResult] 
╔═════════════════════════════════════════════
║ Process LocationResult
╠═════════════════════════════════════════════

05-29 10:03:53.307 DEBUG [TSLocationManager calculateMedianAccuracy] Median accuracy: 10.048
05-29 10:03:53.318 INFO [TrackingService a] 
  ℹ️  Distance from stoppedAtLocation: -24.03905
05-29 10:03:53.320 INFO [b persist] 
  ✅  INSERT: 30d74ab2-6dc9-4663-b8a3-8069fac7ccf8
05-29 10:04:48.765 DEBUG [TrackingService a] 
╔═════════════════════════════════════════════
║ TrackingService: LocationResult
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884529,147.329852 hAcc=17 et=+8d19h54m10s314ms alt=11.100000381469727 vAcc=2 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 83ms, time: 1559088288677

05-29 10:04:48.770 DEBUG [TSLocationManager onLocationResult] 
╔═════════════════════════════════════════════
║ Process LocationResult
╠═════════════════════════════════════════════

05-29 10:04:48.775 DEBUG [TSLocationManager calculateMedianAccuracy] Median accuracy: 9.762
05-29 10:04:48.789 INFO [TrackingService a] 
  ℹ️  Distance from stoppedAtLocation: -32.114647
05-29 10:04:48.791 INFO [b persist] 
  ✅  INSERT: 5a48a152-a182-4cc1-aaf9-d30e42b6c023
05-29 10:08:41.532 INFO [ScheduleEvent onOneShot] 
╔═════════════════════════════════════════════
║ ⏰ OneShot event fired: STOP_TIMEOUT
╠═════════════════════════════════════════════

05-29 10:08:41.564 DEBUG [TSConfig c] ℹ️   Persist config, dirty: [isMoving]
05-29 10:08:41.585 INFO [TSLocationManager removeLocationUpdates] 
  🔴  Location-services: OFF
05-29 10:08:41.594 INFO [TSGeofenceManager stopMonitoringSignificantLocationChanges] 
  🔴  Stop monitoring significant location changes
05-29 10:08:41.598 INFO [HeartbeatService a] 
  🔴  Stop heartbeat
05-29 10:08:41.606 DEBUG [b a] 
  ℹ️  LocationAuthorization: Permission granted
05-29 10:08:41.615 INFO [TrackingService a] 
  🔵  setPace: true → false
05-29 10:08:41.685 INFO [LocationRequestService b] 
  ℹ️  Location availability: true
05-29 10:08:42.512 INFO [LocationRequestService a] 
╔═════════════════════════════════════════════
║ motionchange LocationResult: 9
╠═════════════════════════════════════════════
╟─ 📍  Location[fused -42.884526,147.329835 hAcc=18 et=+8d19h58m4s81ms alt=10.800000190734863 vAcc=2 sAcc=??? bAcc=??? {Bundle[mParcelledData.dataSize=52]}], age: 66ms, time: 1559088522443

05-29 10:08:42.518 INFO [TSLocationManager onSingleLocationResult] 
  🔵  Acquired motionchange position, isMoving: false
05-29 10:08:42.520 DEBUG [TSLocationManager calculateMedianAccuracy] Median accuracy: 15.258
05-29 10:08:42.538 INFO [b persist] 
  ✅  INSERT: 17bd4908-0976-4e78-a831-125baebf6e9a
05-29 10:08:42.544 DEBUG [TSGeofenceManager startMonitoringStationaryRegion] 
  🎾  Start monitoring stationary region (radius: 150.0 meters)
05-29 10:08:42.545 INFO [HttpService flush] 
╔═════════════════════════════════════════════
║ HTTP Service (count: 48)
╠═════════════════════════════════════════════

05-29 10:08:42.549 INFO [ActivityRecognitionService a] 
  🎾  Start motion-activity updates
05-29 10:08:42.562 INFO [BackgroundTaskManager onStartJob] ⏳ startBackgroundTask: 10
05-29 10:08:42.597 DEBUG [b allWithLocking] 
  ✅  Locked 48 records
05-29 10:08:42.599 INFO [HttpService a] 
  🔵  HTTP POST batch (48)
05-29 10:08:42.616 INFO [TrackingService f] 
╔═════════════════════════════════════════════
║ TrackingService motionchange: false
╠═════════════════════════════════════════════

05-29 10:08:42.629 DEBUG [ActivityRecognitionService a] 
  🚘 ️DetectedActivity [type=STILL, confidence=100]
05-29 10:08:42.673 DEBUG [TrackingService onDestroy] 
  🔴  TrackingService destroyed
05-29 10:08:42.686 DEBUG [ActivityRecognitionService a] 
  🚘 ️DetectedActivity [type=STILL, confidence=100]
05-29 10:08:42.699 DEBUG [ActivityRecognitionService onDestroy] 
  🔴  ActivityRecognitionService destroyed
05-29 10:08:43.583 WARN [HttpService$b onResponse] 
  ⚠️  Response: 404, Not Found
05-29 10:08:43.592 DEBUG [b unlock] 
  ✅  UNLOCKED (48)
05-29 10:08:43.597 INFO [BackgroundTaskManager$Task stop] ⏳ stopBackgroundTask: 10
05-29 10:08:43.602 DEBUG [BackgroundTaskService onDestroy] 
```
</details>

**Comment by christocracy** [comment]
You can set a very high stopTimeout.

**Comment by christocracy** [comment]
Android devices typically require just a few meters of movement to trigger into the moving state.

**Comment by nattyg93** [comment]
Thanks for getting back so quickly.

Do you consider setting a very high `stopTimeout` the correct way to get this behaviour then? Should we ignore the `disableStopTimeout` docs?

Our experimentation with a wide variety of devices in real life situations has found that it can (at worst) be several hundred meters of walking before tracking begins. As we are in part trying to accurately measure how far people are walking, we are losing up to half the data at times. On lower end devices, it can take even longer before it initiates the tracking. Presumably this is a short coming of the activity detection service on the low end devices, but due to this we have opted to go for a more battery hungry approach for the sake of higher resolution data.

**Comment by christocracy** [comment]
Yes, lower end devices can take longer to engage, particularly those missing sensors, such as gyroscope / magnetometer.

I can’t remember if I actually implemented disableStopDetection for Android or if it just never made it into this year’s big refactor.

It’s pretty trivial to implement for Android and someone else has asked about it, so it will be done.

Setting a very high stopTimeout is perfectly fine (eg 60 min).  It might even be preferable as a safety mechanism.  If the user leaves location-services on for 8 hours straight, you’re not going to get any data because the battery will be dead.

I hope you offer a UI button to let the user #stop the plugin so you don’t consume all their battery.

**Comment by nattyg93** [comment]
Ok. I'll use the `stopTimeout` as a work around for now.

Yes, we are concerned about battery drain from leaving location services. Unfortunately, the requirement for high resolution data has been deemed more important than battery life. Did the 0.3.4 version of this plugin disable location services after a period, or was it essentially the same behaviour as the 1.0.x version with an infinite `stopTimeout`?

**Comment by christocracy** [comment]
>Did the 0.3.4 version of this plugin disable location services after a period

The big changes are documented [here](https://link.medium.com/Sjzy2f8w4W)

The core Philosophy of Operation of the plugin hasn’t changed in 4 years.  For Android, that’s the Motion API triggering location-services on/off with stopTimeout.

**Comment by christocracy** [comment]
Fixed in 1.0.8
